alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java

```diff
@@ -115,13 +120,25 @@ protected NettyChannelBuilder delegate() {
 
   @Override
   public ManagedChannel build() {
-    CheckGcpEnvironment.check(enableUntrustedAlts);
+    Status status = Status.OK;
+    if (!CheckGcpEnvironment.isOnGcp()) {
+      if (enableUntrustedAlts) {
+        logger.log(
+            Level.WARNING,
+            "Untrusted ALTS mode is enabled and we cannot guarantee the trustworthiness of the "
+                + "ALTS handshaker service");
+      } else {
+        status =
+            Status.FAILED_PRECONDITION.withDescription(
```

Comment:
The gRPC library can't use FAILED_PRECONDITION: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md

Reply:
How about INTERNAL? I feel FAILED_PRECONDITION is the right description though.
 "Operation was rejected because the system is not in a state required for the operation's execution."

Reply:
INTERNAL is okay. We can change it later if need be.